File: SDKs/beezup.com/2.0/typescript/src/sdk/models/operations/getrules.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class GetRulesPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=storeId" })
  storeId: string;
}


export class GetRulesRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: GetRulesPathParams;
}


export class GetRulesResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  beezUPCommonErrorResponseMessage?: shared.BeezUpCommonErrorResponseMessage;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  ruleList?: shared.RuleList;
}

File: SDKs/beezup.com/2.0/python/src/sdk/models/operations/getrules.py
import dataclasses
from typing import Optional
from ..shared import beezup_common_errorresponsemessage as shared_beezup_common_errorresponsemessage
from ..shared import rulelist as shared_rulelist


@dataclasses.dataclass
class GetRulesPathParams:
    store_id: str = dataclasses.field(metadata={'path_param': { 'field_name': 'storeId', 'style': 'simple', 'explode': False }})
    

@dataclasses.dataclass
class GetRulesRequest:
    path_params: GetRulesPathParams = dataclasses.field()
    

@dataclasses.dataclass
class GetRulesResponse:
    content_type: str = dataclasses.field()
    status_code: int = dataclasses.field()
    beez_up_common_error_response_message: Optional[shared_beezup_common_errorresponsemessage.BeezUpCommonErrorResponseMessage] = dataclasses.field(default=None)
    rule_list: Optional[shared_rulelist.RuleList] = dataclasses.field(default=None)
    
